x-pack/plugins/reporting/server/config/create_config.ts

```diff
@@ -0,0 +1,124 @@
+import { TypeOf } from '@kbn/config-schema';
+import crypto from 'crypto';
+import { capitalize } from 'lodash';
+import { map, mergeMap } from 'rxjs/operators';
+import { CoreSetup, Logger, PluginInitializerContext } from 'src/core/server';
+import { getDefaultChromiumSandboxDisabled } from './default_chromium_sandbox_disabled';
+import { ConfigSchema } from './schema';
+
+/*
+ * Set up dynamic config defaults
+ * - xpack.capture.browser.chromium.disableSandbox
+ * - xpack.kibanaServer
+ * - xpack.reporting.encryptionKey
+ */
+export function createConfig$(core: CoreSetup, context: PluginInitializerContext, logger: Logger) {
+  return context.config.create<TypeOf<typeof ConfigSchema>>().pipe(
+    map(config => {
+      // encryption key
+      let encryptionKey = config.encryptionKey;
+      if (encryptionKey === undefined) {
+        logger.warn(
+          i18n.translate('xpack.reporting.serverConfig.randomEncryptionKey', {
+            defaultMessage:
+              'Generating a random key for xpack.reporting.encryptionKey. To prevent sessions from being invalidated on ' +
+              'restart, please set xpack.reporting.encryptionKey in kibana.yml',
+          })
+        );
+        encryptionKey = crypto.randomBytes(16).toString('hex');
+      }
+      const { kibanaServer: reportingServer } = config;
+      const serverInfo = core.http.getServerInfo();
+      // kibanaServer.hostname, default to server.host, don't allow "0"
+      let kibanaServerHostname = reportingServer.hostname
+        ? reportingServer.hostname
+        : serverInfo.host;
+      if (kibanaServerHostname === '0') {
+        logger.warn(
+          i18n.translate('xpack.reporting.serverConfig.invalidServerHostname', {
+            defaultMessage:
+              `Found 'server.host: "0"' in Kibana configuration. This is incompatible with Reporting. ` +
+              `To enable Reporting to work, '{configKey}: 0.0.0.0' is being automatically to the configuration. ` +
+              `You can change the setting to 'server.host: 0.0.0.0' or add '{configKey}: 0.0.0.0' in kibana.yml to prevent this message.`,
+            values: { configKey: 'xpack.reporting.kibanaServer.hostname' },
+          })
+        );
+        kibanaServerHostname = '0.0.0.0';
+      }
+      // kibanaServer.port, default to server.port
+      const kibanaServerPort = reportingServer.port
+      ? reportingServer.port
+      : serverInfo.port; // prettier-ignore
+      // kibanaServer.protocol, default to server.protocol
+      const kibanaServerProtocol = reportingServer.protocol
+        ? reportingServer.protocol
+        : serverInfo.protocol;
+      return {
+        ...config,
+        encryptionKey,
+        kibanaServer: {
+          hostname: kibanaServerHostname,
+          port: kibanaServerPort,
+          protocol: kibanaServerProtocol,
+        },
+      };
+    }),
+    mergeMap(async config => {
+      if (config.capture.browser.chromium.disableSandbox != null) {
```

Comment:
Would a config value of `false` here still disable the sandbox?

Reply:
The config value of `false` **should enable** the sandbox; `true` **should disable** the sandbox

I get regularly get confused about it, because the config key refers to a negative.

At this point of the code, we check if the key exists in the user's kibana.yml: a non `undefined` or `null` value indicates that it is, so we can return the config as-is in this callback and leave the user's setting untouched.

Hope that makes sense!